### PR TITLE
✨ feat(cookie-consent-banner): introduce ::part pseudo selector

### PR DESCRIPTION
**Pull request by switchnollie**
This PR adds the possibility to style parts of the web component using the CSS `::part` selector. Thereby we can provide nice looking defaults and for customization either via the easy-to-use CSS variable solution or via the fully flexible `::part()` API.

### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://github.com/porscheofficial/cookie-consent-banner/blob/cd4e241b0929042c3a6592f023ec1dac485e9d22/CONTRIBUTOR_LICENSE_AGREEMENT.md)

### CODE REVIEW

File: packages/cookie-consent-banner/src/components/cookie-consent-banner/cookie-consent-banner.tsx [hunk 1]
```diff
@@ -217,24 +217,28 @@ export class CookieConsentBanner {
...
           )}
           <form>
-            <p class="cc_text">
+            <p class="cc_text" part="description">
               <slot />
             </p>
             {Boolean(this.isShownSettings) && (
               <div class="cc_settings">
-                <p class="cc_settings_description">
+                <p part="description" class="cc_settings_description">
```
Comment (pboeder) [inline packages/cookie-consent-banner/src/components/cookie-consent-banner/cookie-consent-banner.tsx:230]: Is it intentional that the part "description" is there twice?

File: packages/cookie-consent-banner/src/components/cookie-consent-banner/cookie-consent-banner.tsx [hunk 2]
```diff
@@ -280,6 +285,7 @@ export class CookieConsentBanner {
               {!this.isShownSettings &&
                 !!this.btnLabelOnlyEssentialAndContinue && (
                   <button
+                    part="secondary-button"
```
Comment (pboeder) [inline packages/cookie-consent-banner/src/components/cookie-consent-banner/cookie-consent-banner.tsx:288]: I would split the naming of both buttons.

File: packages/cookie-consent-banner/src/components/cookie-consent-banner/cookie-consent-banner.tsx (hunk 1 above)
Comment (switchnollie) [reply packages/cookie-consent-banner/src/components/cookie-consent-banner/cookie-consent-banner.tsx:230]: I guess you're right, let's use `description-settings` and `description-main` separately.

File: packages/cookie-consent-banner/src/components/cookie-consent-banner/cookie-consent-banner.tsx (hunk 2 above)
Comment (switchnollie) [reply packages/cookie-consent-banner/src/components/cookie-consent-banner/cookie-consent-banner.tsx]: Switched to these parts: `button-accept-all`, `button-persist-selection` and `button-essential-only`

**Review by pboeder — APPROVED** [review]

**Pull request merged**